es-index-cleaner test is failing

When running the e2e tests for the 1.9 release, @objectiser and I were having test issues, like:

```
$ make test
Running unit tests...
?   	github.com/jaegertracing/jaeger-operator/cmd	[no test files]
?   	github.com/jaegertracing/jaeger-operator/cmd/manager	[no test files]
ok  	github.com/jaegertracing/jaeger-operator/pkg/account	0.083s	coverage: 100.0% of statements
?   	github.com/jaegertracing/jaeger-operator/pkg/apis	[no test files]
ok  	github.com/jaegertracing/jaeger-operator/pkg/apis/io/v1alpha1	0.012s	coverage: 17.6% of statements
?   	github.com/jaegertracing/jaeger-operator/pkg/cmd/start	[no test files]
?   	github.com/jaegertracing/jaeger-operator/pkg/cmd/version	[no test files]
ok  	github.com/jaegertracing/jaeger-operator/pkg/config/sampling	0.027s	coverage: 94.1% of statements
ok  	github.com/jaegertracing/jaeger-operator/pkg/config/ui	0.018s	coverage: 94.1% of statements
?   	github.com/jaegertracing/jaeger-operator/pkg/controller	[no test files]
?   	github.com/jaegertracing/jaeger-operator/pkg/controller/deployment	[no test files]
ok  	github.com/jaegertracing/jaeger-operator/pkg/controller/jaeger	3.064s	coverage: 64.3% of statements
ok  	github.com/jaegertracing/jaeger-operator/pkg/cronjob	0.032s	coverage: 93.5% of statements
ok  	github.com/jaegertracing/jaeger-operator/pkg/deployment	0.058s	coverage: 100.0% of statements
ok  	github.com/jaegertracing/jaeger-operator/pkg/ingress	0.041s	coverage: 100.0% of statements
ok  	github.com/jaegertracing/jaeger-operator/pkg/inject	0.008s	coverage: 100.0% of statements
ok  	github.com/jaegertracing/jaeger-operator/pkg/route	0.056s	coverage: 100.0% of statements
ok  	github.com/jaegertracing/jaeger-operator/pkg/service	0.007s	coverage: 100.0% of statements
ok  	github.com/jaegertracing/jaeger-operator/pkg/storage	0.005s	coverage: 100.0% of statements
ok  	github.com/jaegertracing/jaeger-operator/pkg/strategy	0.006s	coverage: 100.0% of statements
ok  	github.com/jaegertracing/jaeger-operator/pkg/util	0.005s	coverage: 100.0% of statements
?   	github.com/jaegertracing/jaeger-operator/pkg/version	[no test files]
customresourcedefinition.apiextensions.k8s.io/jaegers.io.jaegertracing created
Formatting code...
Building...
Sending build context to Docker daemon  120.8MB
Step 1/4 : FROM alpine:3.8
 ---> 196d12cf6ab1
Step 2/4 : USER nobody
 ---> Using cache
 ---> 18305c5b670d
Step 3/4 : ADD build/_output/bin/jaeger-operator /usr/local/bin/jaeger-operator
 ---> 7bd39e079480
Step 4/4 : ENTRYPOINT ["/usr/local/bin/jaeger-operator"]
 ---> Running in 93fc80b1586f
Removing intermediate container 93fc80b1586f
 ---> 35a21f78d012
Successfully built 35a21f78d012
Successfully tagged jpkroehling/jaeger-operator:latest
Pushing image jpkroehling/jaeger-operator:latest...
Running end-to-end tests...
time="2019-01-24T10:12:00+01:00" level=info msg="passing &{{Jaeger io.jaegertracing/v1alpha1} {agent-as-daemonset  jaeger-jaeger-group-daemonset-1548321110    0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[] [] nil [] } {allInOne { {map[]} {[] [] map[] {map[] map[]}}} {0  {map[]} {[] [] map[] {map[] map[]}}} {0  {map[]} {[] [] map[] {map[] map[]}}} {DaemonSet  {map[log-level:debug]} {[] [] map[] {map[] map[]}}} {{[]}} {{[]}} {  {map[]} {<nil>   } {false     false  false false false} {false 0  }} {<nil>  {[] [] map[] {map[] map[]}}} {[] [] map[] {map[] map[]}}} {}}"
time="2019-01-24T10:13:27+01:00" level=info msg="passing &{{Jaeger io.jaegertracing/v1alpha1} {with-cassandra  jaeger-jaeger-group-cassandra-1548321201    0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[] [] nil [] } {allInOne { {map[]} {[] [] map[] {map[] map[]}}} {0  {map[]} {[] [] map[] {map[] map[]}}} {0  {map[]} {[] [] map[] {map[] map[]}}} {  {map[]} {[] [] map[] {map[] map[]}}} {{[]}} {{[]}} {cassandra  {map[cassandra.servers:cassandra.default.svc cassandra.keyspace:jaeger_v1_datacenter1]} {<nil>  datacenter1 } {false     false  false false false} {false 0  }} {<nil>  {[] [] map[] {map[] map[]}}} {[] [] map[] {map[] map[]}}} {}}"
Forwarding from 127.0.0.1:16686 -> 16686
Forwarding from [::1]:16686 -> 16686
Forwarding from 127.0.0.1:14268 -> 14268
Forwarding from [::1]:14268 -> 14268

E0124 10:14:17.474440   18677 portforward.go:339] error closing listener: close tcp4 127.0.0.1:16686: use of closed network connection
E0124 10:14:17.475638   18677 portforward.go:339] error closing listener: close tcp6 [::1]:16686: use of closed network connection
E0124 10:14:17.476764   18677 portforward.go:339] error closing listener: close tcp4 127.0.0.1:14268: use of closed network connection
E0124 10:14:17.477881   18677 portforward.go:339] error closing listener: close tcp6 [::1]:14268: use of closed network connection
Forwarding from 127.0.0.1:16686 -> 16686
Forwarding from [::1]:16686 -> 16686
Forwarding from 127.0.0.1:14268 -> 14268
Forwarding from [::1]:14268 -> 14268

E0124 10:18:04.149215   18677 portforward.go:339] error closing listener: close tcp4 127.0.0.1:16686: use of closed network connection
E0124 10:18:04.149273   18677 portforward.go:339] error closing listener: close tcp6 [::1]:16686: use of closed network connection
E0124 10:18:04.150349   18677 portforward.go:339] error closing listener: close tcp4 127.0.0.1:14268: use of closed network connection
E0124 10:18:04.151444   18677 portforward.go:339] error closing listener: close tcp6 [::1]:14268: use of closed network connection
time="2019-01-24T10:18:19+01:00" level=info msg="passing &{{Jaeger io.jaegertracing/v1alpha1} {my-jaeger  jaeger-jaeger-group-my-jaeger-1548321484    0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[] [] nil [] } {allInOne { {map[log-level:debug memory.max-traces:10000]} {[] [] map[] {map[] map[]}}} {0  {map[]} {[] [] map[] {map[] map[]}}} {0  {map[]} {[] [] map[] {map[] map[]}}} {  {map[]} {[] [] map[] {map[] map[]}}} {{[]}} {{[]}} {  {map[]} {<nil>   } {false     false  false false false} {false 0  }} {<nil>  {[] [] map[] {map[] map[]}}} {[] [] map[] {map[] map[]}}} {}}"
time="2019-01-24T10:18:19+01:00" level=info msg="passing &{{Jaeger io.jaegertracing/v1alpha1} {my-jaeger  jaeger-jaeger-group-my-other-jaeger-1548321484    0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[] [] nil [] } {allInOne { {map[log-level:debug memory.max-traces:10000]} {[] [] map[] {map[] map[]}}} {0  {map[]} {[] [] map[] {map[] map[]}}} {0  {map[]} {[] [] map[] {map[] map[]}}} {  {map[]} {[] [] map[] {map[] map[]}}} {{[]}} {{[]}} {  {map[]} {<nil>   } {false     false  false false false} {false 0  }} {<nil>  {[] [] map[] {map[] map[]}}} {[] [] map[] {map[] map[]}}} {}}"
Forwarding from 127.0.0.1:16686 -> 16686
Forwarding from [::1]:16686 -> 16686

Forwarding from 127.0.0.1:14268 -> 14268
Forwarding from [::1]:14268 -> 14268

E0124 10:18:50.245753   18677 portforward.go:339] error closing listener: close tcp4 127.0.0.1:14268: use of closed network connection
E0124 10:18:50.245805   18677 portforward.go:339] error closing listener: close tcp6 [::1]:14268: use of closed network connection
E0124 10:18:50.245782   18677 portforward.go:339] error closing listener: close tcp4 127.0.0.1:16686: use of closed network connection
E0124 10:18:50.247958   18677 portforward.go:339] error closing listener: close tcp6 [::1]:16686: use of closed network connection
--- FAIL: TestJaeger (449.81s)
    --- FAIL: TestJaeger/jaeger-group (373.71s)
        --- FAIL: TestJaeger/jaeger-group/es-index-cleaner (85.87s)
            client.go:57: resource type RoleBinding with namespace/name (jaeger-jaeger-group-es-index-cleaner-1548321398/jaeger-operator) created
            client.go:57: resource type Role with namespace/name (jaeger-jaeger-group-es-index-cleaner-1548321398/jaeger-operator) created
            client.go:57: resource type ServiceAccount with namespace/name (jaeger-jaeger-group-es-index-cleaner-1548321398/jaeger-operator) created
            client.go:57: resource type Deployment with namespace/name (jaeger-jaeger-group-es-index-cleaner-1548321398/jaeger-operator) created
            jaeger_test.go:60: Initialized cluster resources. Namespace: jaeger-jaeger-group-es-index-cleaner-1548321398
            wait_util.go:51: Deployment available (1/1)
            client.go:57: resource type Jaeger with namespace/name (jaeger-jaeger-group-es-index-cleaner-1548321398/test-es-index-cleaner) created
            wait_util.go:51: Deployment available (1/1)
            wait_util.go:161: Waiting for conjob test-es-index-cleaner-es-index-cleaner to have scheduled
            wait_util.go:167: CronJob succeeded
            wait_util.go:136: Waiting for job of owner test-es-index-cleaner-es-index-cleaner to succeed.
            wait_util.go:136: Waiting for job of owner test-es-index-cleaner-es-index-cleaner to succeed.
            wait_util.go:136: Waiting for job of owner test-es-index-cleaner-es-index-cleaner to succeed.
            wait_util.go:136: Waiting for job of owner test-es-index-cleaner-es-index-cleaner to succeed.
            wait_util.go:136: Waiting for job of owner test-es-index-cleaner-es-index-cleaner to succeed.
            wait_util.go:136: Waiting for job of owner test-es-index-cleaner-es-index-cleaner to succeed.
            wait_util.go:136: Waiting for job of owner test-es-index-cleaner-es-index-cleaner to succeed.
            wait_util.go:136: Waiting for job of owner test-es-index-cleaner-es-index-cleaner to succeed.
            wait_util.go:136: Waiting for job of owner test-es-index-cleaner-es-index-cleaner to succeed.
            wait_util.go:136: Waiting for job of owner test-es-index-cleaner-es-index-cleaner to succeed.
            wait_util.go:136: Waiting for job of owner test-es-index-cleaner-es-index-cleaner to succeed.
            wait_util.go:136: Waiting for job of owner test-es-index-cleaner-es-index-cleaner to succeed.
            wait_util.go:136: Waiting for job of owner test-es-index-cleaner-es-index-cleaner to succeed.
            es_index_cleaner_test.go:19: timed out waiting for the condition
            client.go:75: resource type Jaeger with namespace/name (jaeger-jaeger-group-es-index-cleaner-1548321398/test-es-index-cleaner) successfully deleted
            client.go:75: resource type Deployment with namespace/name (jaeger-jaeger-group-es-index-cleaner-1548321398/jaeger-operator) successfully deleted
            client.go:75: resource type ServiceAccount with namespace/name (jaeger-jaeger-group-es-index-cleaner-1548321398/jaeger-operator) successfully deleted
            client.go:75: resource type Role with namespace/name (jaeger-jaeger-group-es-index-cleaner-1548321398/jaeger-operator) successfully deleted
            client.go:75: resource type RoleBinding with namespace/name (jaeger-jaeger-group-es-index-cleaner-1548321398/jaeger-operator) successfully deleted
FAIL
FAIL	github.com/jaegertracing/jaeger-operator/test/e2e	449.882s
make: *** [Makefile:75: e2e-tests] Error 1
```

According to `git bisect`, this was introduced in commit from Dec 5:

```
    9adb6a288b02e05e780f2474e191bcc8114ec363 is the first bad commit
    commit 9adb6a288b02e05e780f2474e191bcc8114ec363
    Author: Pavol Loffay <p.loffay@gmail.com>
    Date:   Wed Dec 5 21:35:44 2018 +0100
     
        Add Elasticsearch index cleaner as cron job (#155)
       
        * Add Elasticsearch index cleaner
       
        Signed-off-by: Pavol Loffay <ploffay@redhat.com>
       
        * Fix review comments
       
        Signed-off-by: Pavol Loffay <ploffay@redhat.com>
     
    :040000 040000 5710ac76e9b613abd7885098e4427ac8c817a19e 60364ecb0af687d86d6e805e31e7df6f30b098fd M      deploy
    :040000 040000 d6a8e93204248ee0d7309eb80bbb67a5bfcfd727 5ade03d46edf1772d4025aafae7727d21cb40df8 M      pkg
    :040000 040000 aff1595f2ba9d32a9a5de92e0011da64c193494b 44066bf48b5ff6e7b077ebe98509f380e794d41f M      test
```

--- a/test/e2e/suite_elasticsearch_test.go
+++ b/test/e2e/suite_elasticsearch_test.go
@@ -30,6 +30,6 @@ func TestElasticsearch(t *testing.T) {
 	t.Run("elasticsearch", func(t *testing.T) {
 		t.Run("spark-dependencies-es", SparkDependenciesElasticsearch)
 		t.Run("simple-prod", SimpleProd)
-		// t.Run("es-index-cleaner", EsIndexCleaner)
+		t.Run("es-index-cleaner", EsIndexCleaner)
 	})
 }

